treeherder/webapp/api/performance_data.py

```diff
@@ -64,10 +64,24 @@ def list(self, request, project):
                 framework__in=frameworks)
 
         interval = request.query_params.get('interval')
+        start_date = request.query_params.get('start_date')  # 'YYYY-MM-DDTHH:MM:SS
+        end_date = request.query_params.get('end_date')  # 'YYYY-MM-DDTHH:MM:SS'
+
         if interval:
+            if start_date:
+                return Response({"message": "Provide either interval only -or- start (and end) date"},
+                                status=HTTP_400_BAD_REQUEST)
+            else:
+                signature_data = signature_data.filter(
+                    last_updated__gte=datetime.datetime.fromtimestamp(
+                        int(time.time() - int(interval))))
+
+        if start_date and end_date:
             signature_data = signature_data.filter(
-                last_updated__gte=datetime.datetime.fromtimestamp(
-                    int(time.time() - int(interval))))
+                last_updated__gte=start_date, last_updated__lte=end_date)
+        elif start_date:
+            signature_data = signature_data.filter(
+                last_updated__gte=start_date)
```

Comment:
You could just chain filter operations like this:

```py
if start_date:
   signature_data = signature_data.filter(last_updated__gte=start_date)
if end_date:
   signature_data = signature_data.filter(last_updated__lte=end_date)
```

Django is smart and won't actually convert this into a database query until you actually request the objects.